docs/Code_Quality_Guidelines.rst

```diff
@@ -0,0 +1,68 @@
+.. _Code_Quality_Guidelines:
+
+*******************
+TARDIS Coding Guide
+*******************
+
+General Code Format
+===================
+
+TARDIS follows the `Black <https://black.readthedocs.io/en/stable/>`_ coding style for files, and files submitted should follow the Black style. To install Black, do::
+
+    pip install black
+    
+For more information on Black formatting, please refer to `Black <https://black.readthedocs.io/en/stable/>`_. 
+
+Edge Cases and Exception Handling
+=================================
+
+Code should be written to handle errors that can occur during the run. From `Astropy's code guide: <https://docs.astropy.org/en/latest/development/codeguide.html#standard-output-warnings-and-errors>`_ "For errors/exceptions, one should always use raise with one of the built-in exception classes, or a custom exception class. The nondescript Exception class should be avoided as much as possible, in favor of more specific exceptions (IOError, ValueError, etc.)". By handling exceptions and errors the code can handle and alert the user of invalid input, and with error messages the code can inform the user where the error is coming from and how to avoid the error. An example of this in practice is taken from the `kromer plot widget by Jaladh Singhal <https://github.com/tardis-sn/tardis/blob/master/tardis/widgets/kromer_plot.py#L447-L451>`_::
+
+    if packets_mode not in ["virtual", "real"]:
+            raise ValueError(
+                "Invalid value passed to packets_mode. Only "
+                "allowed values are 'virtual' or 'real'"
+            )
+In this example of code, any invalid input triggers the conditional statement, and it raises a specific error (ValueError) and gives a message that informs the user what their error is and the appropriate values that can be used to avoid this error. In this manner, most edge case inputs and possible errors can be caught so the user can address them. This practice allows for specific errors to be avoided to make debugging easier and makes it so edge cases of invalid input are caught.
+
+Docstrings
+==========
+
+For docstrings, TARDIS follows the docstring formatting of `Numpydocs <https://numpydoc.readthedocs.io/en/latest/format.html>`_. 
+This is an example of a properly formatted docstring::
+
+    """
+    Read different density file formats
```

Comment:
We should also add `def function_name():` here to indicate where docstring lives